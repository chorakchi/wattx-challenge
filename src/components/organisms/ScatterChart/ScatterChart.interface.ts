import { HTMLAttributes } from "react";

export interface ScatterChartProps extends HTMLAttributes<HTMLElement> {
  items: any;
}
