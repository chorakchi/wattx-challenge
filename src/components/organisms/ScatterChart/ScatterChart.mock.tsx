export const mockData = {
  items: [
    {
      z: -1.21,
      x: 1142753026162,
      y: 34298045435,
      name: "Bitcoin",
      symbol: "BTC",
    },
    {
      z: -1.1,
      x: 449085638512,
      y: 16584143779,
      name: "Ethereum",
      symbol: "ETH",
    },
    {
      z: -1.85,
      x: 78131460865,
      y: 1916181083,
      name: "Binance Coin",
      symbol: "BNB",
    },
    {
      z: -1.83,
      x: 71547533741,
      y: 2445734567,
      name: "Cardano",
      symbol: "ADA",
    },
    {
      z: 0.05,
      x: 68763150434,
      y: 65732725908,
      name: "Tether",
      symbol: "USDT",
    },
    {
      z: -0.39,
      x: 53171017204,
      y: 3856792966,
      name: "XRP",
      symbol: "XRP",
    },
    {
      z: -2.21,
      x: 47265907114,
      y: 1653521652,
      name: "Solana",
      symbol: "SOL",
    },
    {
      z: -4.59,
      x: 40950443340,
      y: 2303739906,
      name: "Polkadot",
      symbol: "DOT",
    },
    {
      z: 0.01,
      x: 32767597855,
      y: 2889731799,
      name: "USD Coin",
      symbol: "USDC",
    },
    {
      z: 0.31,
      x: 31111983877,
      y: 1553161170,
      name: "Dogecoin",
      symbol: "DOGE",
    },
    {
      z: 3.65,
      x: 16391847175,
      y: 511130000,
      name: "Uniswap",
      symbol: "UNI",
    },
    {
      z: -1.54,
      x: 14630746575,
      y: 555524188,
      name: "Terra",
      symbol: "LUNA",
    },
    {
      z: -0.93,
      x: 13093855974,
      y: 328767974,
      name: "Wrapped Bitcoin",
      symbol: "WBTC",
    },
    {
      z: 0.01,
      x: 13072916791,
      y: 5122729614,
      name: "Binance USD",
      symbol: "BUSD",
    },
    {
      z: -2.06,
      x: 12775926011,
      y: 2803320964,
      name: "Litecoin",
      symbol: "LTC",
    },
    {
      z: 3.08,
      x: 12644853313,
      y: 719781037,
      name: "Avalanche",
      symbol: "AVAX",
    },
    {
      z: 1,
      x: 12497136779,
      y: 1124081200,
      name: "Chainlink",
      symbol: "LINK",
    },
    {
      z: -0.05,
      x: 11822726051,
      y: 1351462551,
      name: "Bitcoin Cash",
      symbol: "BCH",
    },
    {
      z: 2.16,
      x: 11030032497,
      y: 353191371,
      name: "Algorand",
      symbol: "ALGO",
    },
    {
      z: -1.47,
      x: 10088489037,
      y: 1574039844,
      name: "Polygon",
      symbol: "MATIC",
    },
    {
      z: -0.57,
      x: 10042074961,
      y: 1927416934,
      name: "SHIBA INU",
      symbol: "SHIB",
    },
    {
      z: 10.74,
      x: 9665496993,
      y: 1880176117,
      name: "Stellar",
      symbol: "XLM",
    },
    {
      z: 5.96,
      x: 7949057749,
      y: 538669637,
      name: "Internet Computer",
      symbol: "ICP",
    },
    {
      z: -1.71,
      x: 7665040117,
      y: 407145456,
      name: "VeChain",
      symbol: "VET",
    },
    {
      z: -1.82,
      x: 7578422103,
      y: 483247362,
      name: "Axie Infinity",
      symbol: "AXS",
    },
    {
      z: -1.8,
      x: 7531451989,
      y: 545611389,
      name: "Filecoin",
      symbol: "FIL",
    },
    {
      z: -0.09,
      x: 7349047665,
      y: 417350174,
      name: "Cosmos",
      symbol: "ATOM",
    },
    {
      z: -0.83,
      x: 7054398935,
      y: 1095995879,
      name: "TRON",
      symbol: "TRX",
    },
    {
      z: -0.81,
      x: 7046378144,
      y: 788312362,
      name: "Ethereum Classic",
      symbol: "ETC",
    },
    {
      z: -1.39,
      x: 6751345624,
      y: 180538628,
      name: "FTX Token",
      symbol: "FTT",
    },
    {
      z: -0.04,
      x: 6472571590,
      y: 477616232,
      name: "Dai",
      symbol: "DAI",
    },
    {
      z: -1.09,
      x: 6365375720,
      y: 92854736,
      name: "Bitcoin BEP2",
      symbol: "BTCB",
    },
    {
      z: -2.09,
      x: 6122733847,
      y: 177810462,
      name: "THETA",
      symbol: "THETA",
    },
    {
      z: 0.47,
      x: 5965577121,
      y: 263515915,
      name: "Tezos",
      symbol: "XTZ",
    },
    {
      z: 1.92,
      x: 5505978647,
      y: 242402087,
      name: "Hedera",
      symbol: "HBAR",
    },
    {
      z: 8.65,
      x: 5392340055,
      y: 477447373,
      name: "Fantom",
      symbol: "FTM",
    },
    {
      z: -6.23,
      x: 4980005434,
      y: 285297775,
      name: "Flow",
      symbol: "FLOW",
    },
    {
      z: 1.42,
      x: 4825719828,
      y: 32994993,
      name: "Crypto.com Coin",
      symbol: "CRO",
    },
    {
      z: -3.64,
      x: 4794047428,
      y: 186099190,
      name: "Monero",
      symbol: "XMR",
    },
    {
      z: 1.33,
      x: 4714345078,
      y: 96463763,
      name: "Elrond",
      symbol: "EGLD",
    },
    {
      z: -0.39,
      x: 4592811613,
      y: 261003519,
      name: "PancakeSwap",
      symbol: "CAKE",
    },
    {
      z: -2.56,
      x: 4379985159,
      y: 1052818146,
      name: "EOS",
      symbol: "EOS",
    },
    {
      z: -0.35,
      x: 4090516047,
      y: 149781567,
      name: "NEAR Protocol",
      symbol: "NEAR",
    },
    {
      z: 1.23,
      x: 4020393799,
      y: 220826559,
      name: "Aave",
      symbol: "AAVE",
    },
    {
      z: -2.63,
      x: 3977658421,
      y: 97175619,
      name: "Klaytn",
      symbol: "KLAY",
    },
    {
      z: -1.62,
      x: 3853322459,
      y: 148852514,
      name: "eCash",
      symbol: "XEC",
    },
    {
      z: 11.2,
      x: 3847530294,
      y: 64472556,
      name: "Quant",
      symbol: "QNT",
    },
    {
      z: 1.13,
      x: 3759243234,
      y: 377242787,
      name: "The Graph",
      symbol: "GRT",
    },
    {
      z: -0.76,
      x: 3584611922,
      y: 73752626,
      name: "IOTA",
      symbol: "MIOTA",
    },
    {
      z: -0.67,
      x: 3234077849,
      y: 290632877,
      name: "Bitcoin SV",
      symbol: "BSV",
    },
  ],
};
