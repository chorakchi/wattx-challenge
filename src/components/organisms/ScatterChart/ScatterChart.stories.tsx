import { ComponentStory, ComponentMeta } from "@storybook/react";

import { ScatterChart } from "./ScatterChart";
import {mockData } from './ScatterChart.mock'

export default {
  title: "Organisms/ScatterChart",
  component: ScatterChart,
} as ComponentMeta<typeof ScatterChart>;

const Template: ComponentStory<typeof ScatterChart> = (args) => (
  <ScatterChart {...args} />
);

export const ScatterChart_ = Template.bind({});
ScatterChart_.args = {
  ...mockData
}