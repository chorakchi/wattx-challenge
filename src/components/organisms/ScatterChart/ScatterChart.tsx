import { StyledWrapper } from "./ScatterChart.styled";
import { ScatterChartProps } from "./ScatterChart.interface";
import Highcharts from "highcharts";
import HighchartsReact from "highcharts-react-official";
import HighchartMore from "highcharts/highcharts-more";
import React, { useState } from "react";
import { optionsProvider } from "./ScatterChart.helper";

export const ScatterChart: React.FC<ScatterChartProps> = ({
  items,
  ...props
}: ScatterChartProps) => {
  const [highchartMoreLoaded, setHighchartMoreLoaded] = useState(false);
  setTimeout(() => {
    HighchartMore(Highcharts);
    setHighchartMoreLoaded(true);
  }, 0);

  return (
    <StyledWrapper hidden= {!items.length}>
      {highchartMoreLoaded && (
        <HighchartsReact {...props} highcharts={Highcharts} options={optionsProvider(items)} />
      )}
    </StyledWrapper>
  );
};
