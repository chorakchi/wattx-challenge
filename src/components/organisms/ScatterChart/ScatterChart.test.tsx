import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { ScatterChart } from "./ScatterChart";
import { mockData } from "./ScatterChart.mock";

jest.mock("components/atoms/SVG/SVG", () => ({
  SVG: "svg",
}));

describe("ScatterChart", () => {
  it("should render the ScatterChart", () => {
    const subject = shallow(
      <ScatterChart {...mockData}/>
    );
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer
      .create(
        <ScatterChart {...mockData}/>
      )
      .toJSON();
    expect(subject).toMatchSnapshot();
  });
});
