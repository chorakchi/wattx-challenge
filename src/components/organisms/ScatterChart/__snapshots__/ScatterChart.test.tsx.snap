// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScatterChart should render correctly and match to the Snapshot 1`] = `
<div
  className="Boxstyled__BoxStyled-sc-12nhtob-0 bOPHwF ScatterChartstyled__StyledWrapper-sc-1yjqzm4-0 DfZoV"
  hidden={false}
/>
`;
